app/models/benefitting_country.rb

```diff
@@ -0,0 +1,40 @@
+class BenefittingCountry
+  include ActiveModel::Model
+  attr_accessor :name, :recipient_code, :code, :regions
+
+  class << self
+    def all
+      @all ||= Codelist.new(type: "benefitting_countries", source: "beis").map { |country| new_from_hash(country) }
```

Comment:
This is my confusion: would this `all` be calculated on every request, or would it get memoised on application load?

I remember something about codelists being loaded on application load (in production environments), for efficiency. Would the same happen to this codelist?

Reply:
Yeah, it'll be saved in memory on the first usage

Reply:
Yep, so it's memoized twice, once on the load of all the codelists and again when the `Codelist` class instance is instaniated, which I think is fine